src/vs/workbench/api/common/extHostFileSystem.ts

```diff
@@ -115,6 +115,7 @@ export class ExtHostFileSystem implements ExtHostFileSystemShape {
 	private readonly _fsProvider = new Map<number, vscode.FileSystemProvider>();
 	private readonly _registeredSchemes = new Set<string>();
 	private readonly _watches = new Map<number, IDisposable>();
+	private readonly _enableProposedApi = new Map<number, boolean>();
```

Comment:
not a super-fan of this extra data structure (which doesn't seem to be clean up btw). could be consider debt tho and cleaned up later